File: python/2.7/factory/ChicagoPizzaIngredientFactory.py
from PizzaIngredientFactory import *

from ThickCrustDough import *
from PlumTomatoSauce import *
from MozzarellaCheese import *
from BlackOlives import *
from Spinach import *
from Eggplant import *
from SlicedPepperoni import *
from FrozenClams import *

class ChicagoPizzaIngredientFactory(PizzaIngredientFactory):
    def __init__(self):
        super(ChicagoPizzaIngredientFactory, self).__init__()

    def createDough(self):
        return str(ThickCrustDough())

    def createSauce(self):
        return str(PlumTomatoSauce())

    def createCheese(self):
        return str(MozzarellaCheese())

    def createVeggies(self):
        self.veggies = [str(BlackOlives()), str(Spinach()), str(Eggplant())]
        return self.veggies

    def createPepperoni(self):
        return str(SlicedPepperoni())

    def createClam(self):
        return str(FrozenClams())

File: js/node/factory/com/pizza/pizzaIngredientFactory/ChicagoPizzaIngredientFactory.js
require('./PizzaIngredientFactory');

require('../dough/ThickCrustDough');
require('../sauce/PlumTomatoSauce');
require('../cheese/MozzarellaCheese');
require('../veggies/BlackOlives');
require('../veggies/Spinach');
require('../veggies/Eggplant');
require('../pepperoni/SlicedPepperoni');
require('../clams/FrozenClams');

ChicagoPizzaIngredientFactory = function() {
    PizzaIngredientFactory.apply(this);
}
ChicagoPizzaIngredientFactory.prototype = new PizzaIngredientFactory();
ChicagoPizzaIngredientFactory.prototype.createDough = function() {
    return new ThickCrustDough();
}
ChicagoPizzaIngredientFactory.prototype.createSauce = function() {
    return new PlumTomatoSauce();
}
ChicagoPizzaIngredientFactory.prototype.createCheese = function() {
    return new MozzarellaCheese();
}
ChicagoPizzaIngredientFactory.prototype.createVeggies = function() {
    this.veggies = [];
    this.veggies.push(new BlackOlives());
    this.veggies.push(new Spinach());
    this.veggies.push(new Eggplant());
    return this.veggies;
}
ChicagoPizzaIngredientFactory.prototype.createPepperoni = function() {
    return new SlicedPepperoni();
}
ChicagoPizzaIngredientFactory.prototype.createClam = function() {
    return new FrozenClams();
}